File: src/de/umr/fixcon/graphFunctions/standardFunctions/HasNoTrianglesFunction.java
package de.umr.fixcon.graphFunctions.standardFunctions;

import com.google.common.graph.Graph;
import de.umr.fixcon.graphFunctions.DecisionProblem;
import de.umr.fixcon.graphFunctions.GraphFunction;

import static de.umr.core.GraphAlgorithmsKt.hasTriangle;

public class HasNoTrianglesFunction extends DecisionProblem implements GraphFunction {
    @Override
    public boolean isEdgeMonotone() {
        return false;
    }

    @Override
    public double apply(final Graph<Integer> g, final int... parameters) {
        return (!hasTriangle(g)) ? 1 : 0;
    }
}

File: src/de/umr/fixcon/graphFunctions/standardFunctions/HasNoTrianglesFunction.kt
package de.umr.fixcon.graphFunctions.standardFunctions

import com.google.common.graph.Graph
import de.umr.core.hasTriangle
import de.umr.fixcon.graphFunctions.GraphFunction

class HasNoTrianglesFunction :  GraphFunction {
    override val isEdgeMonotone: Boolean = false

    override fun apply(g: Graph<Int>, args: List<Int>): Double = if (!hasTriangle(g)) 1.0 else 0.0
}